app/models/model_mixin.py

```diff
@@ -76,3 +75,37 @@ def get_by_id(cls, id):
     def toDict(self):
         return {
             c.key: getattr(self, c.key) for c in inspect(self).mapper.column_attrs}
+
+
+    def graph_data(self, start, end, set_by):
```

Comment:
It should be a class method, meaning you should do something like this
```suggestion
    @classmethod
    def graph_data(cls, start, end, set_by):
```